core/gotham_state.rs

```diff
@@ -11,28 +12,41 @@ use std::collections::BTreeMap;
 
 #[derive(Default)]
 pub struct GothamState {
+  #[cfg(not(feature = "unstable_type_name"))]
   data: BTreeMap<TypeId, Box<dyn Any>>,
+  #[cfg(feature = "unstable_type_name")]
+  data: BTreeMap<&'static str, Box<dyn Any>>,
```

Comment:
This might have an impact on runtime perf -- can we use typeid in release to avoid string compares?

Reply:
Yeah, the feature is only for dev release builds (disabled by default). I benched http and websocket, there is not any measurable perf hit - most of it is avoided by using the constant version of `type_name` with the `const_type_name` unstable feature.